modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala

```diff
@@ -76,12 +76,13 @@ final class NurtureAlg[F[_]](
       updates <- sbtAlg.getUpdatesForRepo(repo)
       filtered <- filterAlg.localFilterMany(repoConfig, updates)
       grouped = Update.group(filtered)
-      _ <- logger.info(util.logger.showUpdates(grouped))
+      limited = repoConfig.updatesLimit.fold(grouped)(grouped.take)
```

Comment:
@manuelcueto I think the limit is applied too early here. `limited` could contain an update for which a PR exists that was closed without merging. IMO the limit should only take updates into account that create or update a PR.

Reply:
Agree. I thought create or update were the only choices so it seemed valid here. I'll take a look.

Reply:
If `processUpdate` would return an enum that describes what it has done (created a PR, updated a PR, nothing) instead of `Unit`, it would be easier to apply the limit here, I guess.

Reply:
Got to a similar solution before reading this 😅